src/katsdpmodels/primary_beam.py

```diff
@@ -730,27 +733,38 @@ def sample(self, l: ArrayLike, m: ArrayLike, frequency: u.Quantity,   # noqa: E7
             return self._finalize(jones, output_type, out=out)
 
     @staticmethod
-    @numba.njit
     def _sample_grid_impl(x_m: np.ndarray, y_m: np.ndarray,
                           l: np.ndarray, m: np.ndarray,
                           wavenumber: np.ndarray,
                           samples: np.ndarray,
                           out: np.ndarray) -> None:
+        # This function was originally split into a static method so that it
+        # could be optimised with numba. However, when sampling a large grid it
+        # is actually significantly faster without numba. If numpy is using
+        # OpenBLAS then it also parallelises the matrix multiplications and so
+        # there is no advantage to using numba parallelisation.
+        tmp = np.zeros((len(y_m), len(l) * 4), np.complex64)
+        coeff1 = None  # Reuse the memory for these from one loop to the next
+        coeff2 = None
         for freq_idx in np.ndindex(wavenumber.shape):
             x = x_m * wavenumber[freq_idx]
             y = y_m * wavenumber[freq_idx]
-            coeff1 = _expjm2pi(np.outer(l, x))
-            coeff2 = _expjm2pi(np.outer(m, y))
+            coeff1 = _expjm2pi(np.outer(l, x), out=coeff1)
```

Comment:
Where does `_expjm2pi` support the `out` parameter?

Reply:
`numba.vectorize` provides it (it creates a numpy ufunc, and all ufuncs support a whole range of keyword arguments).